// @myparcel-eslint/eslint-config-typescript/dist/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  default: () => src_default2
});
module.exports = __toCommonJS(src_exports);
var import_eslint = require("eslint");

// ../eslint-config/src/index.ts
var defaultConfig = {
  env: {
    browser: true
  },
  rules: {
    "accessor-pairs": "off",
    "array-bracket-newline": [
      "warn",
      "consistent"
    ],
    "array-bracket-spacing": [
      "warn",
      "never"
    ],
    "array-callback-return": "warn",
    "array-element-newline": [
      "warn",
      "consistent"
    ],
    "block-scoped-var": "warn",
    "block-spacing": "warn",
    "brace-style": [
      "warn",
      "1tbs"
    ],
    "callback-return": "off",
    camelcase: [
      "warn",
      {
        ignoreDestructuring: true,
        ignoreImports: true,
        properties: "never"
      }
    ],
    "capitalized-comments": "off",
    "class-methods-use-this": "warn",
    "comma-dangle": [
      "warn",
      "always-multiline"
    ],
    "comma-spacing": [
      "warn",
      {
        before: false,
        after: true
      }
    ],
    "comma-style": "warn",
    complexity: [
      "warn",
      {
        max: 10
      }
    ],
    "computed-property-spacing": [
      "warn",
      "never"
    ],
    "consistent-return": "off",
    "consistent-this": [
      "warn",
      "self"
    ],
    curly: "warn",
    "default-case": "off",
    "default-case-last": "warn",
    "default-param-last": "warn",
    "dot-location": [
      "warn",
      "property"
    ],
    "dot-notation": "warn",
    "eol-last": "warn",
    eqeqeq: [
      "warn",
      "smart"
    ],
    "for-direction": "warn",
    "func-call-spacing": [
      "warn",
      "never"
    ],
    "func-name-matching": "off",
    "func-names": "off",
    "func-style": [
      "warn",
      "declaration",
      {
        allowArrowFunctions: true
      }
    ],
    "function-call-argument-newline": [
      "warn",
      "consistent"
    ],
    "function-paren-newline": [
      "warn",
      "consistent"
    ],
    "getter-return": "off",
    "global-require": "off",
    "grouped-accessor-pairs": "warn",
    "guard-for-in": "off",
    "handle-callback-err": "off",
    "id-blacklist": "off",
    "id-denylist": "off",
    "id-length": [
      "warn",
      {
        exceptions: [
          "$",
          "_",
          "i",
          "e",
          "a",
          "b",
          "x",
          "y",
          "z"
        ]
      }
    ],
    "id-match": "off",
    "implicit-arrow-linebreak": [
      "warn",
      "beside"
    ],
    indent: [
      "warn",
      2,
      {
        SwitchCase: 1
      }
    ],
    "indent-legacy": "off",
    "init-declarations": "off",
    "jsx-quotes": "off",
    "key-spacing": "warn",
    "keyword-spacing": [
      "warn",
      {
        before: true,
        after: true
      }
    ],
    "line-comment-position": [
      "warn",
      {
        position: "above"
      }
    ],
    "linebreak-style": [
      "warn",
      "unix"
    ],
    "lines-around-comment": [
      "warn",
      {
        afterBlockComment: false,
        afterLineComment: false,
        allowArrayEnd: false,
        allowArrayStart: true,
        allowBlockEnd: false,
        allowBlockStart: true,
        allowClassEnd: false,
        allowClassStart: true,
        allowObjectEnd: false,
        allowObjectStart: true,
        beforeBlockComment: true
      }
    ],
    "lines-around-directive": "warn",
    "lines-between-class-members": [
      "warn",
      "always",
      {
        exceptAfterSingleLine: true
      }
    ],
    "max-classes-per-file": "warn",
    "max-depth": [
      "warn",
      {
        max: 3
      }
    ],
    "max-len": [
      "warn",
      120
    ],
    "max-lines": "off",
    "max-lines-per-function": [
      "warn",
      {
        max: 50,
        skipComments: true
      }
    ],
    "max-nested-callbacks": [
      "warn",
      {
        max: 3
      }
    ],
    "max-params": [
      "warn",
      4
    ],
    "max-statements": "off",
    "max-statements-per-line": "warn",
    "multiline-comment-style": "off",
    "multiline-ternary": [
      "warn",
      "always-multiline"
    ],
    "new-cap": [
      "warn"
    ],
    "new-parens": "warn",
    "newline-after-var": "off",
    "newline-before-return": "off",
    "newline-per-chained-call": "off",
    "no-alert": "warn",
    "no-array-constructor": "warn",
    "no-bitwise": "warn",
    "no-buffer-constructor": "warn",
    "no-caller": "warn",
    "no-case-declarations": "warn",
    "no-catch-shadow": "warn",
    "no-compare-neg-zero": "warn",
    "no-cond-assign": "warn",
    "no-console": "warn",
    "no-constant-condition": "warn",
    "no-constructor-return": "warn",
    "no-continue": "warn",
    "no-control-regex": "off",
    "no-debugger": "error",
    "no-delete-var": "warn",
    "no-div-regex": "off",
    "no-dupe-args": "warn",
    "no-dupe-else-if": "warn",
    "no-dupe-keys": "warn",
    "no-duplicate-case": "warn",
    "no-else-return": "warn",
    "no-empty": "warn",
    "no-empty-character-class": "warn",
    "no-empty-function": "warn",
    "no-empty-pattern": "warn",
    "no-eq-null": "warn",
    "no-eval": "warn",
    "no-ex-assign": "warn",
    "no-extend-native": "warn",
    "no-extra-bind": "warn",
    "no-extra-boolean-cast": "warn",
    "no-extra-label": "warn",
    "no-extra-parens": "warn",
    "no-extra-semi": "warn",
    "no-fallthrough": "warn",
    "no-floating-decimal": "warn",
    "no-func-assign": "warn",
    "no-global-assign": "warn",
    "no-implicit-coercion": [
      "warn",
      {
        boolean: false
      }
    ],
    "no-implicit-globals": "warn",
    "no-implied-eval": "warn",
    "no-import-assign": "warn",
    "no-inline-comments": "warn",
    "no-inner-declarations": "off",
    "no-invalid-regexp": "warn",
    "no-invalid-this": "warn",
    "no-irregular-whitespace": "warn",
    "no-iterator": "warn",
    "no-label-var": "warn",
    "no-labels": "off",
    "no-lone-blocks": "off",
    "no-lonely-if": "warn",
    "no-loop-func": "off",
    "no-loss-of-precision": "warn",
    "no-magic-numbers": [
      "warn",
      {
        ignore: [
          -1,
          0,
          1,
          100,
          1e3
        ],
        detectObjects: true
      }
    ],
    "no-misleading-character-class": "off",
    "no-mixed-operators": "warn",
    "no-mixed-requires": "off",
    "no-mixed-spaces-and-tabs": "warn",
    "no-multi-assign": "warn",
    "no-multi-spaces": "warn",
    "no-multi-str": "off",
    "no-multiple-empty-lines": [
      "warn",
      {
        max: 1,
        maxBOF: 0,
        maxEOF: 0
      }
    ],
    "no-native-reassign": "off",
    "no-negated-condition": "warn",
    "no-negated-in-lhs": "off",
    "no-nested-ternary": "warn",
    "no-new": "off",
    "no-new-func": "off",
    "no-new-object": "warn",
    "no-new-require": "off",
    "no-new-wrappers": "off",
    "no-nonoctal-decimal-escape": "warn",
    "no-obj-calls": "off",
    "no-octal": "off",
    "no-octal-escape": "off",
    "no-param-reassign": "off",
    "no-path-concat": "off",
    "no-plusplus": "off",
    "no-process-env": "off",
    "no-process-exit": "off",
    "no-promise-executor-return": "warn",
    "no-proto": "off",
    "no-prototype-builtins": "off",
    "no-redeclare": "off",
    "no-regex-spaces": "warn",
    "no-restricted-exports": "warn",
    "no-restricted-globals": "warn",
    "no-restricted-modules": "warn",
    "no-restricted-properties": "warn",
    "no-restricted-syntax": "warn",
    "no-return-assign": "warn",
    "no-script-url": "off",
    "no-self-assign": "off",
    "no-self-compare": "off",
    "no-sequences": "warn",
    "no-setter-return": "warn",
    "no-shadow": "off",
    "no-shadow-restricted-names": "off",
    "no-spaced-func": "off",
    "no-sparse-arrays": "off",
    "no-sync": "off",
    "no-tabs": "warn",
    "no-template-curly-in-string": "warn",
    "no-ternary": "off",
    "no-throw-literal": "off",
    "no-trailing-spaces": "warn",
    "no-undef": "warn",
    "no-undef-init": "off",
    "no-undefined": "off",
    "no-underscore-dangle": [
      "warn",
      {
        allowAfterThis: true
      }
    ],
    "no-unexpected-multiline": "warn",
    "no-unmodified-loop-condition": "warn",
    "no-unneeded-ternary": "warn",
    "no-unreachable": "warn",
    "no-unreachable-loop": "warn",
    "no-unsafe-finally": "off",
    "no-unsafe-negation": "off",
    "no-unsafe-optional-chaining": "warn",
    "no-unused-expressions": "warn",
    "no-unused-labels": "warn",
    "no-unused-private-class-members": "warn",
    "no-unused-vars": [
      "warn",
      {
        ignoreRestSiblings: true
      }
    ],
    "no-use-before-define": "off",
    "no-useless-backreference": "warn",
    "no-useless-call": "warn",
    "no-useless-catch": "warn",
    "no-useless-concat": "warn",
    "no-useless-escape": "warn",
    "no-useless-return": "warn",
    "no-void": "off",
    "no-warning-comments": "off",
    "no-whitespace-before-property": "warn",
    "no-with": "warn",
    "nonblock-statement-body-position": "off",
    "object-curly-newline": [
      "warn",
      {
        ObjectExpression: {
          consistent: true
        },
        ObjectPattern: {
          consistent: true
        },
        ImportDeclaration: {
          consistent: true
        },
        ExportDeclaration: "never"
      }
    ],
    "object-curly-spacing": [
      "warn",
      "never"
    ],
    "object-property-newline": [
      "warn",
      {
        allowAllPropertiesOnSameLine: true
      }
    ],
    "one-var": [
      "warn",
      "never"
    ],
    "one-var-declaration-per-line": [
      "warn",
      "always"
    ],
    "operator-assignment": "warn",
    "operator-linebreak": [
      "warn",
      "before"
    ],
    "padded-blocks": [
      "warn",
      {
        blocks: "never",
        switches: "never",
        classes: "never"
      }
    ],
    "padding-line-between-statements": [
      "warn",
      {
        blankLine: "always",
        prev: "cjs-import",
        next: "*"
      },
      {
        blankLine: "never",
        prev: "cjs-import",
        next: "cjs-import"
      },
      {
        blankLine: "always",
        prev: "*",
        next: "cjs-export"
      },
      {
        blankLine: "never",
        prev: "cjs-export",
        next: "cjs-export"
      },
      {
        blankLine: "always",
        prev: "export",
        next: "*"
      },
      {
        blankLine: "always",
        prev: "export",
        next: "export"
      },
      {
        blankLine: "always",
        prev: "import",
        next: "*"
      },
      {
        blankLine: "never",
        prev: "import",
        next: "import"
      },
      {
        blankLine: "always",
        prev: "*",
        next: "block-like"
      },
      {
        blankLine: "always",
        prev: "block-like",
        next: "*"
      },
      {
        blankLine: "always",
        prev: "*",
        next: "block"
      },
      {
        blankLine: "always",
        prev: "block",
        next: "*"
      },
      {
        blankLine: "always",
        prev: "*",
        next: "if"
      },
      {
        blankLine: "always",
        prev: "if",
        next: "*"
      }
    ],
    "prefer-exponentiation-operator": "warn",
    "prefer-named-capture-group": "off",
    "prefer-object-spread": "warn",
    "prefer-promise-reject-errors": "off",
    "prefer-reflect": "off",
    "prefer-regex-literals": "warn",
    "quote-props": [
      "warn",
      "as-needed"
    ],
    quotes: [
      "warn",
      "single"
    ],
    radix: [
      "warn",
      "as-needed"
    ],
    "require-atomic-updates": "off",
    "require-unicode-regexp": "off",
    semi: [
      "warn",
      "always"
    ],
    "semi-spacing": "warn",
    "semi-style": [
      "warn",
      "last"
    ],
    "sort-keys": "off",
    "sort-vars": "off",
    "space-before-blocks": [
      "warn",
      "always"
    ],
    "space-before-function-paren": [
      "warn",
      "never"
    ],
    "space-in-parens": [
      "warn",
      "never"
    ],
    "space-infix-ops": "warn",
    "space-unary-ops": [
      "warn",
      {
        words: true,
        nonwords: false
      }
    ],
    "spaced-comment": [
      "warn",
      "always",
      {
        block: {
          balanced: true
        },
        markers: [
          "/"
        ]
      }
    ],
    strict: "off",
    "switch-colon-spacing": "warn",
    "template-tag-spacing": "off",
    "unicode-bom": "off",
    "use-isnan": "off",
    "valid-typeof": "warn",
    "vars-on-top": "warn",
    "wrap-iife": "off",
    "wrap-regex": "warn",
    yoda: [
      "warn",
      "never"
    ]
  }
};
var src_default = defaultConfig;

// src/index.ts
var import_eslint_config_es6 = __toESM(require("@myparcel-eslint/eslint-config-es6"));
var overrides = {
  "brace-style": "off",
  camelcase: "off",
  "comma-dangle": "off",
  "comma-spacing": "off",
  "func-call-spacing": "off",
  indent: "off",
  "no-array-constructor": "off",
  "no-dupe-class-members": "off",
  "no-empty-function": "off",
  "no-extra-parens": "off",
  "no-extra-semi": "off",
  "no-implied-eval": "off",
  "no-magic-numbers": "off",
  "no-throw-literal": "off",
  "no-undef": "off",
  "no-unused-expressions": "off",
  "no-unused-vars": "off",
  "no-use-before-define": "off",
  "no-useless-constructor": "off",
  "padding-line-between-statements": "off",
  quotes: "off",
  "require-await": "off",
  semi: "off",
  "space-before-function-paren": "off"
};
var typescriptConfig = {
  extends: [
    require.resolve("@myparcel-eslint/eslint-config-esnext"),
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended"
  ],
  parser: "@typescript-eslint/parser",
  parserOptions: {
    project: "tsconfig.json",
    tsconfigRootDir: "."
  },
  plugins: [
    "@typescript-eslint"
  ],
  rules: {
    ...overrides,
    "@typescript-eslint/adjacent-overload-signatures": "warn",
    "@typescript-eslint/array-type": "warn",
    "@typescript-eslint/await-thenable": "warn",
    "@typescript-eslint/ban-ts-comment": "warn",
    "@typescript-eslint/ban-tslint-comment": "off",
    "@typescript-eslint/ban-types": "warn",
    "@typescript-eslint/brace-style": src_default.rules["brace-style"],
    "@typescript-eslint/class-literal-property-style": "off",
    "@typescript-eslint/comma-dangle": src_default.rules["comma-dangle"],
    "@typescript-eslint/comma-spacing": src_default.rules["comma-spacing"],
    "@typescript-eslint/consistent-indexed-object-style": "warn",
    "@typescript-eslint/consistent-type-assertions": "warn",
    "@typescript-eslint/consistent-type-definitions": "off",
    "@typescript-eslint/consistent-type-exports": "off",
    "@typescript-eslint/consistent-type-imports": "off",
    "@typescript-eslint/default-param-last": "warn",
    "@typescript-eslint/dot-notation": "off",
    "@typescript-eslint/explicit-function-return-type": "warn",
    "@typescript-eslint/explicit-member-accessibility": "warn",
    "@typescript-eslint/explicit-module-boundary-types": "warn",
    "@typescript-eslint/func-call-spacing": src_default.rules["func-call-spacing"],
    "@typescript-eslint/indent": src_default.rules.indent,
    "@typescript-eslint/init-declarations": "off",
    "@typescript-eslint/keyword-spacing": "off",
    "@typescript-eslint/lines-between-class-members": "off",
    "@typescript-eslint/member-delimiter-style": "warn",
    "@typescript-eslint/member-ordering": [
      "warn",
      {
        default: [
          "public-field",
          "protected-field",
          "private-field",
          "public-get",
          "protected-get",
          "private-get",
          "public-set",
          "protected-set",
          "private-set",
          "public-method",
          "protected-method",
          "private-method"
        ]
      }
    ],
    "@typescript-eslint/padding-line-between-statements": [
      "warn",
      {
        blankLine: "always",
        prev: "export",
        next: "*"
      },
      {
        blankLine: "always",
        prev: "export",
        next: "export"
      },
      {
        blankLine: "always",
        prev: "import",
        next: "*"
      },
      {
        blankLine: "never",
        prev: "import",
        next: "import"
      },
      {
        blankLine: "always",
        prev: "block-like",
        next: "*"
      },
      {
        blankLine: "always",
        prev: "*",
        next: "block"
      },
      {
        blankLine: "always",
        prev: "block",
        next: "*"
      },
      {
        blankLine: "always",
        prev: "*",
        next: "if"
      },
      {
        blankLine: "always",
        prev: "if",
        next: "*"
      },
      {
        blankLine: "always",
        prev: "*",
        next: [
          "interface",
          "type"
        ]
      },
      {
        blankLine: "always",
        prev: [
          "interface",
          "type"
        ],
        next: "*"
      }
    ],
    "@typescript-eslint/method-signature-style": "off",
    "@typescript-eslint/naming-convention": [
      "warn",
      {
        selector: "default",
        format: [
          "strictCamelCase"
        ],
        filter: {
          regex: "^_$",
          match: false
        }
      },
      {
        selector: "variable",
        format: [
          "strictCamelCase",
          "UPPER_CASE"
        ],
        filter: {
          regex: "^_$",
          match: false
        }
      },
      {
        selector: "property",
        format: [
          "camelCase",
          "PascalCase"
        ]
      },
      {
        selector: [
          "classProperty",
          "objectLiteralProperty",
          "typeProperty",
          "classMethod",
          "objectLiteralMethod",
          "typeMethod",
          "accessor",
          "enumMember"
        ],
        format: null,
        modifiers: [
          "requiresQuotes"
        ]
      },
      {
        selector: [
          "objectLiteralProperty"
        ],
        format: null
      },
      {
        selector: "typeLike",
        format: [
          "StrictPascalCase"
        ]
      },
      {
        selector: "enumMember",
        format: [
          "UPPER_CASE"
        ]
      },
      {
        selector: "property",
        modifiers: [
          "private"
        ],
        format: [
          "strictCamelCase"
        ],
        leadingUnderscore: "allow"
      },
      {
        selector: "typeParameter",
        format: null
      }
    ],
    "@typescript-eslint/no-array-constructor": src_default.rules["no-array-constructor"],
    "@typescript-eslint/no-base-to-string": "off",
    "@typescript-eslint/no-confusing-non-null-assertion": "off",
    "@typescript-eslint/no-confusing-void-expression": "off",
    "@typescript-eslint/no-dupe-class-members": "off",
    "@typescript-eslint/no-duplicate-imports": "off",
    "@typescript-eslint/no-dynamic-delete": "warn",
    "@typescript-eslint/no-empty-function": src_default.rules["no-empty-function"],
    "@typescript-eslint/no-empty-interface": "warn",
    "@typescript-eslint/no-explicit-any": "warn",
    "@typescript-eslint/no-extra-non-null-assertion": "warn",
    "@typescript-eslint/no-extra-parens": src_default.rules["no-extra-parens"],
    "@typescript-eslint/no-extra-semi": src_default.rules["no-extra-semi"],
    "@typescript-eslint/no-extraneous-class": "warn",
    "@typescript-eslint/no-floating-promises": "warn",
    "@typescript-eslint/no-for-in-array": "warn",
    "@typescript-eslint/no-implicit-any-catch": "off",
    "@typescript-eslint/no-implied-eval": src_default.rules["no-implied-eval"],
    "@typescript-eslint/no-inferrable-types": "warn",
    "@typescript-eslint/no-invalid-this": "off",
    "@typescript-eslint/no-invalid-void-type": "off",
    "@typescript-eslint/no-loop-func": "off",
    "@typescript-eslint/no-loss-of-precision": "off",
    "@typescript-eslint/no-magic-numbers": [
      "warn",
      {
        ...src_default.rules["no-magic-numbers"][1],
        ignoreEnums: true,
        ignoreNumericLiteralTypes: true,
        ignoreReadonlyClassProperties: true
      }
    ],
    "@typescript-eslint/no-meaningless-void-operator": "off",
    "@typescript-eslint/no-misused-new": "warn",
    "@typescript-eslint/no-misused-promises": "warn",
    "@typescript-eslint/no-namespace": "warn",
    "@typescript-eslint/no-non-null-asserted-nullish-coalescing": "off",
    "@typescript-eslint/no-non-null-asserted-optional-chain": "warn",
    "@typescript-eslint/no-non-null-assertion": "warn",
    "@typescript-eslint/no-parameter-properties": "warn",
    "@typescript-eslint/no-redeclare": "off",
    "@typescript-eslint/no-require-imports": "off",
    "@typescript-eslint/no-restricted-imports": "off",
    "@typescript-eslint/no-shadow": "off",
    "@typescript-eslint/no-this-alias": "warn",
    "@typescript-eslint/no-throw-literal": src_default.rules["no-throw-literal"],
    "@typescript-eslint/no-type-alias": "off",
    "@typescript-eslint/no-unnecessary-boolean-literal-compare": "off",
    "@typescript-eslint/no-unnecessary-condition": "off",
    "@typescript-eslint/no-unnecessary-qualifier": "warn",
    "@typescript-eslint/no-unnecessary-type-arguments": "warn",
    "@typescript-eslint/no-unnecessary-type-assertion": "warn",
    "@typescript-eslint/no-unnecessary-type-constraint": "off",
    "@typescript-eslint/no-unsafe-argument": "off",
    "@typescript-eslint/no-unsafe-assignment": "off",
    "@typescript-eslint/no-unsafe-call": "off",
    "@typescript-eslint/no-unsafe-member-access": "off",
    "@typescript-eslint/no-unsafe-return": "off",
    "@typescript-eslint/no-unused-expressions": src_default.rules["no-unused-expressions"],
    "@typescript-eslint/no-unused-vars": src_default.rules["no-unused-vars"],
    "@typescript-eslint/no-use-before-define": src_default.rules["no-use-before-define"],
    "@typescript-eslint/no-useless-constructor": import_eslint_config_es6.default.rules["no-useless-constructor"],
    "@typescript-eslint/no-var-requires": "warn",
    "@typescript-eslint/non-nullable-type-assertion-style": "off",
    "@typescript-eslint/object-curly-spacing": "off",
    "@typescript-eslint/prefer-as-const": "off",
    "@typescript-eslint/prefer-enum-initializers": "off",
    "@typescript-eslint/prefer-for-of": "warn",
    "@typescript-eslint/prefer-function-type": "warn",
    "@typescript-eslint/prefer-includes": "warn",
    "@typescript-eslint/prefer-literal-enum-member": "off",
    "@typescript-eslint/prefer-namespace-keyword": "warn",
    "@typescript-eslint/prefer-nullish-coalescing": "warn",
    "@typescript-eslint/prefer-optional-chain": "warn",
    "@typescript-eslint/prefer-readonly": "off",
    "@typescript-eslint/prefer-readonly-parameter-types": "off",
    "@typescript-eslint/prefer-reduce-type-parameter": "off",
    "@typescript-eslint/prefer-regexp-exec": "warn",
    "@typescript-eslint/prefer-return-this-type": "off",
    "@typescript-eslint/prefer-string-starts-ends-with": "warn",
    "@typescript-eslint/prefer-ts-expect-error": "off",
    "@typescript-eslint/promise-function-async": "off",
    "@typescript-eslint/quotes": src_default.rules.quotes,
    "@typescript-eslint/require-array-sort-compare": "off",
    "@typescript-eslint/require-await": import_eslint_config_es6.default.rules["require-await"],
    "@typescript-eslint/restrict-plus-operands": "warn",
    "@typescript-eslint/restrict-template-expressions": "off",
    "@typescript-eslint/return-await": "warn",
    "@typescript-eslint/semi": src_default.rules.semi,
    "@typescript-eslint/sort-type-union-intersection-members": "off",
    "@typescript-eslint/space-before-function-paren": [
      "warn",
      {
        anonymous: "never",
        named: "never",
        asyncArrow: "always"
      }
    ],
    "@typescript-eslint/space-infix-ops": "off",
    "@typescript-eslint/strict-boolean-expressions": "off",
    "@typescript-eslint/switch-exhaustiveness-check": "off",
    "@typescript-eslint/triple-slash-reference": "warn",
    "@typescript-eslint/type-annotation-spacing": "warn",
    "@typescript-eslint/typedef": [
      "warn",
      {
        arrowParameter: false,
        objectDestructuring: false,
        variableDeclarationIgnoreFunction: true
      }
    ],
    "@typescript-eslint/unbound-method": "off",
    "@typescript-eslint/unified-signatures": "warn"
  },
  settings: {
    jsdoc: {
      mode: "typescript"
    }
  }
};
var src_default2 = typescriptConfig;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {});
