// @myparcel-eslint/eslint-config-es6/dist/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  default: () => src_default
});
module.exports = __toCommonJS(src_exports);
var es6Config = {
  extends: require.resolve("@myparcel-eslint/eslint-config"),
  parserOptions: {
    ecmaVersion: 6
  },
  rules: {
    "func-names": [
      "warn",
      "as-needed"
    ],
    "no-magic-numbers": [
      "warn",
      {
        ignore: [
          -1,
          0,
          1,
          100
        ],
        detectObjects: true,
        enforceConst: true
      }
    ],
    "arrow-body-style": "off",
    "arrow-parens": "warn",
    "arrow-spacing": [
      "warn",
      {
        before: true,
        after: true
      }
    ],
    "constructor-super": "off",
    "generator-star-spacing": "off",
    "no-async-promise-executor": "warn",
    "no-await-in-loop": "off",
    "no-class-assign": "off",
    "no-confusing-arrow": "warn",
    "no-const-assign": "warn",
    "no-dupe-class-members": "warn",
    "no-duplicate-imports": "warn",
    "no-new-symbol": "off",
    "no-restricted-imports": "off",
    "no-return-await": "warn",
    "no-this-before-super": "off",
    "no-useless-computed-key": "off",
    "no-useless-constructor": "off",
    "no-useless-rename": "warn",
    "no-var": "warn",
    "object-shorthand": "off",
    "prefer-arrow-callback": "off",
    "prefer-const": "warn",
    "prefer-destructuring": [
      "warn",
      {
        VariableDeclarator: {
          array: false,
          object: true
        },
        AssignmentExpression: {
          array: false,
          object: false
        }
      }
    ],
    "prefer-numeric-literals": "off",
    "prefer-rest-params": "off",
    "prefer-spread": "warn",
    "prefer-template": "warn",
    "require-await": "warn",
    "require-yield": "off",
    "rest-spread-spacing": "off",
    "sort-imports": "warn",
    "symbol-description": "off",
    "template-curly-spacing": "warn",
    "yield-star-spacing": "off"
  }
};
var src_default = es6Config;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {});
